Transport saga concurrency behavior documentation (#4528)
Particular/docs.particular.net: persistence/azure-storage/index.md

The way to address them is go through the upgrade guide linked above.

> ### Saga concurrency behavior

Author: ```suggestion
## Saga concurrency behavior
```

> Azure Storage Persistence applies [optimistic concurrency control](https://en.wikipedia.org/wiki/Optimistic_concurrency_control) when saga instance are updated concurrently. When this happens the following error will be visible in the log:

Author: ```suggestion
Azure Storage Persistence applies [optimistic concurrency control](https://en.wikipedia.org/wiki/Optimistic_concurrency_control) when saga instances are updated. The persister does not support locking. Optimistic concurrency control conflicts require the message to be reprocessed via recoverability.

Please read the guidance on [saga concurrency](/nservicebus/sagas/concurrency.md) on potential improvements.
```

> 

Author: ```suggestion
### Concurrent access to non-existing saga instances
```

```
NServiceBus.Persistence.AzureStorage.RetryNeededException: This operation requires a retry as it wasn't possible to successfully process it now.
```

> 

Author: ```suggestion
### Concurrent access to existing saga instances
```
